1. Когда используется критерий Стьюдента, а когда Z –критерий? 

1) Если математическое ожидание равно определенному числу, наблюдается нормальное распределение и известно
среднее квадратичное отклонение генеральной совокупности, то используется Z –критерий.
    
2) Если математическое ожидание равно определенному числу, наблюдается нормальное распределение и неизвестно
среднее квадратичное отклонение генеральной совокупности, то используется t(Стьюдента) –критерий.

##### 2.Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

In [1]:
import numpy as np

In [2]:
sd = np.sqrt(4)
sd

2.0

In [3]:
z_h = (17.5 - 17) / (sd / np.sqrt(100))
z_h

2.5

In [4]:
from scipy import stats

In [5]:
stats.norm.cdf(2.5, loc=17, scale=0.6)

2.4942057003208898e-129

In [6]:
critical_value = stats.norm.ppf(1 - 0.05)
critical_value

1.6448536269514722

Гипотиза Н0 отвергается со значением alpha = 0,05 (их средний диаметр не 17 мм)

3. Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.) 

In [7]:
data = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])


In [8]:
st = np.std(data, ddof=1)

In [9]:
st

4.453463071962462

In [10]:
data_mean =  np.mean(data)
data_mean

198.5

In [11]:
t_h = (198.5 - 200) / (st / np.sqrt(10))
t_h

-1.0651074037450896

In [12]:
n = len(data)
n

10

In [13]:
stats.t.ppf(0.01 / 2 , df = n - 1)

-3.24983554401537

t_h > -3.24983554401537 \
Гипотиза Н0 __не__ отвергается со значением alpha = 0,01 (средний вес пачки печенья составляет 200 г)

4. Есть ли статистически значимые различия в росте дочерей? Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165 Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [16]:
mother = np.array([172, 177, 158, 170, 178,175, 164, 160, 169, 165])
dother = np.array([73, 175, 162, 174, 175, 168, 155, 170, 160])
stats.ttest_ind(mother, dother)


Ttest_indResult(statistic=1.142367378626931, pvalue=0.26913616861454137)

In [17]:
alpha = 0.05

pvalue > alpha
Гипотиза Н0 не отвергается со значением alpha = 0,05 (нет статистически значимые различия в росте дочерей)